lib/SIL/SILVerifier.cpp

```diff
@@ -577,6 +578,38 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     SILLocation::LocationKind LocKind = L.getKind();
     ValueKind InstKind = I->getKind();
 
+    // Check that there is at most one debug variable defined
+    // for each argument slot.
+    if (!DS->InlinedCallSite) {
+      SILDebugVariable VarInfo;
+      if (auto *DI = dyn_cast<AllocStackInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      } else if (auto *DI = dyn_cast<AllocBoxInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      } else if (auto *DI = dyn_cast<DebugValueInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      } else if (auto *DI = dyn_cast<DebugValueAddrInst>(I)) {
+        VarInfo = DI->getVarInfo();
+      }
+
+      auto ArgNo = VarInfo.ArgNo;
+      if (ArgNo) {
+        // It is a function argument.
+        if (ArgNo < DebugVars.size() && DebugVars[ArgNo].ArgNo) {
+          require(DebugVars[ArgNo].Name == VarInfo.Name,
+                  "Re-definition of a debug variable");
+        }
+        if (ArgNo >= DebugVars.size()) {
+          // Reserve enough space.
+          unsigned Limit = std::max((size_t)ArgNo, I->getFunction()->getArguments().size());
```

Comment:
This is probably not even necessary. The number of SIL arguments (which could be exploded) is not a good predictor for the number of source arguments, and if we run over the "small" size, we can just let SmallVector handle the allocation (which AFAIK will double the size anyway).

Instead, I would just add filler arguments until we reach ArgNo.